### PR TITLE
mds: Using stl min | max, MIN | MAX macros instead

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
separate commit for reviewer(s).

File: src/mds/MDBalancer.cc [hunk 1]
```diff
@@ -447,7 +447,7 @@ double MDBalancer::try_match(balance_state_t& state, mds_rank_t ex, double& maxe
 {
   if (maxex <= 0 || maxim <= 0) return 0.0;
 
-  double howmuch = MIN(maxex, maxim);
+  double howmuch = std::min(maxex, maxim);
```
Comment (amitkumar50) [inline src/mds/MDBalancer.cc]: PR:https://github.com/ceph/ceph/pull/18124
changes std::min to MIN, std::max to MAX .. 
Why we need changing them back?

File: src/mds/Server.cc [hunk 2]
```diff
@@ -4251,7 +4251,7 @@ void Server::do_open_truncate(MDRequestRef& mdr, int cmode)
   pi->mtime = pi->ctime = mdr->get_op_stamp();
   pi->change_attr++;
 
-  uint64_t old_size = MAX(pi->size, mdr->client_request->head.args.open.old_size);
+  uint64_t old_size = std::max<long unsigned>(pi->size, mdr->client_request->head.args.open.old_size);
```
Comment (batrick) [inline src/mds/Server.cc]: `std::max<uint64_t>`?

File: src/mds/MDBalancer.cc (hunk 1 above)
Comment (batrick) [reply src/mds/MDBalancer.cc]: @amitkumar50, this PR is not "changing them back"? It seems #18124 missed these?

File: src/mds/Server.cc (hunk 2 above)
Comment (shinobu-x) [reply src/mds/Server.cc]: that's better...

**Pull request merged**